.github/workflows/build.yml

```diff
@@ -144,58 +78,98 @@ jobs:
           restore-keys: |
-            ${{ matrix.config.name }}-ccache-
-
-      # Set up the paths for ccache and control the size under 400MB
-      - name: Configure ccache
-        shell: cmake -P {0}
-        run: |
-          file(TO_CMAKE_PATH ${{runner.workspace}} ccache_basedir)
-          set(ENV{CCACHE_BASEDIR} "${ccache_basedir}")
-          set(ENV{CCACHE_DIR} "${ccache_basedir}/.ccache")
-          set(ENV{CCACHE_COMPRESS} "true")
-          set(ENV{CCACHE_COMPRESSLEVEL} "6")
-          set(ENV{CCACHE_MAXSIZE} "400M")
-
-          execute_process(COMMAND ccache -p)
-          execute_process(COMMAND ccache -z)
+            ${{ matrix.config.cc }}-ccache-
 
       - name: Configure CMake
         # Use a bash shell so we can use the same syntax for environment variable
         # access regardless of the host operating system
         shell: bash
-        working-directory: ${{runner.workspace}}/build
-        # Note the current convention is to use the -S and -B options here to specify source 
-        # and build directories, but this is only available with CMake 3.13 and higher.  
+        working-directory: build
+        # Note the current convention is to use the -S and -B options here to specify source
+        # and build directories, but this is only available with CMake 3.13 and higher.
         # The CMake binaries on the Github Actions machines are (as of this writing) 3.12
+        #
         run: |
-          export CC=${{ matrix.config.cc }}
-          export CXX=${{ matrix.config.cxx }}
-          cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+          ccache -p
+          ccache -z
+          cmake .. -DCMAKE_BUILD_TYPE=$BUILD_TYPE
 
       - name: Build
-        working-directory: ${{runner.workspace}}/build
+        working-directory: build
         shell: bash
         # Execute the build.  You can specify a specific target with "--target <NAME>"
         run: |
-          cmake --build . --config $BUILD_TYPE 
+          cmake --build . --config $BUILD_TYPE
 
       # Check the cache size and see if it is over the limit
       - name: Check ccache size
-        shell: cmake -P {0}
+        run: ccache -s
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        if: ${{ matrix.config.cc == 'gcc-8'}}
+        with:
+          name: openfpga
+          path: |
+            abc/abc
+            abc/libabc.a
+            ace2/ace
+            ace2/libace.a
+            openfpga/libopenfpga.a
+            openfpga/openfpga
+            vpr/libvpr.a
+            vpr/vpr
+            yosys/share/
+            yosys/yosys
+            yosys/yosys-abc
+            yosys/yosys-config
+            yosys/yosys-filterlib
+            yosys/yosys-smtbmc
+  regression_tests:
+    name: Regression tests
+    runs-on: ubuntu-18.04
+    container: ghcr.io/lnis-uofu/openfpga-buildenv-test
```

Comment:
I wonder how this `ghcr.io/lnis-uofu/openfpga-buildenv-test` image will update in case there is a change in source code and regression test both.

Reply:
I've added in a new workflow to publish the images. It can be triggered manually from the main branch to update the images. There's no versioning right now of the images, it's always latest. If environment changes in a backwards incompatible way, that's something to add later.